Code/Framework/AzCore/AzCore/Debug/Budget.cpp

```diff
@@ -54,18 +66,63 @@ namespace AZ::Debug
         }
     }
 
-    // TODO:Budgets Methods below are stubbed pending future work to both update budget data and visualize it
-
     void Budget::PerFrameReset()
     {
+        // This is currently called at the beginning of every frame for all budgets.
+        // It has been suggested to call this at more appropriate locations for different budgets.
+        // When that change is made, we would need to ensure that each budget's
+        // per-frame reset is called from the correct thread when a budget needs to be measured
+        // across multiple threads.
```

Comment:
If we're not going to be tracking budget per-frame totals across multiple threads, this can be simplified.